src/coreclr/vm/jitinterface.cpp

```diff
@@ -10776,7 +10776,16 @@ void CEECodeGenInfo::getHelperFtn(CorInfoHelpFunc    ftnNum,               /* IN
 
     accessType = IAT_VALUE;
     targetAddr = (LPVOID)VolatileLoad(&hlpFuncEntryPoints[ftnNum]);
-    if (targetAddr == NULL)
+    if (targetAddr != NULL)
+    {
+        if (helperDef.IsDynamicHelper(&dynamicFtnNum))
+        {
+            (void)LoadDynamicJitHelper(dynamicFtnNum, &helperMD);
```

Comment:
I assume that this call does not actually load anything that was not loaded already, and it is here just to make the assert on the next line possible. Is that right right?

I would rather move the assert next to `targetAddr = (LPVOID)pfnHelper;`. in the slow path. 

Alternative, this should be under ifdef DEBUG so that it is not slowing down the fast path where we are just returning the cache value. There needs to be a comment that we do not expect LoadDynamicJitHelper to load anything that was not loaded already.

Reply:
> I assume that this call does not actually load anything that was not loaded already, and it is here just to make the assert on the next line possible. Is that right right?

It doesn't load anything because it shouldn't, but it is needed since the caller is looking for the `MethodDesc` in some cases. Look before at the `else` case when `targetAddr == NULL`. The dynamic helper may or may not have an associated `MethodDesc`. We need to perform the same logic twice since the `targetAddr` may be a `PortableEntryPoint` or it may be a native function. The only way to detect that is through `LoadDynamicJitHelper`, but that does have a fast out.

Reply:
I've added some additional checks and a comment.

Reply:
Should we add
```
MethodDec* GetMethodDescForILBasedDynamicJitHelper(DynamicCorInfoHelpFunc fncNum)
{
    _ASSERTE(HasILBasedDynamicJitHelper(fncNum));
    return CoreLibBinder::GetMethod(hlpDynamicToBinderMap[ftnNum]);
}
```
and delete the `MethodDesc*` out argument for `LoadDynamicJitHelper` to make this less confusing?

Reply:
There are still scenarios were we want to call load and get the IL. Alternatively, we can call the load and then in the few places that need it call `HasILBasedDynamicJitHelper()`, followed by `GetMethodDescForILBasedDynamicJitHelper()`, would that be your preference?